src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -339,9 +339,14 @@ public PossiblyConditionalState Clone()
             // of analysis based on this choice.
             var rootTemp = BoundDagTemp.ForOriginalInput(expression);
             int originalInputSlot = MakeSlot(expression);
+            var expressionTypeWithAnnotations = expressionTypeWithState.ToTypeWithAnnotations(compilation);
             if (originalInputSlot <= 0)
             {
-                originalInputSlot = makeDagTempSlot(expressionType.ToTypeWithAnnotations(compilation), rootTemp);
+                originalInputSlot = makeDagTempSlot(expressionTypeWithAnnotations, rootTemp);
+                if (!IsConditionalState)
```

Comment:
>if (!IsConditionalState) [](http://example.com/codeflow?start=16&length=24)

Are we covering the alternative branch with tests here, or do we assume the condition is always true?

Reply:
Yes, the alternative branch is covered. We hit it with `true switch { ... }`. You should also be able to hit it with an invocation involving some nullability attributes.